<a href="https://colab.research.google.com/github/Magar-Bhuwan/ML-Internship-Assignments/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

### My lane:
Product Performance. I chose this lane because businesses need to know which products contribute the most to revenue. Understanding product performance helps managers decide which products to promote, restock, or discontinue, making it a useful business question.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Decision
Which products should receive more inventory and marketing investment?

### Action
Increase inventory and marketing for consistently high-performing products, and review low-performing products for improvement or discontinuation.

### Cost of a Wrong Call
Choosing the wrong products could lead to excess inventory, lost sales, and reduced profit because popular products may become unavailable while slow-moving products occupy storage space.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [2]:
from google.colab import files
uploaded = files.upload()

Saving Sales_Dataset_5000_Rows.csv to Sales_Dataset_5000_Rows.csv


In [4]:
import pandas as pd

data = pd.read_csv('Sales_Dataset_5000_Rows.csv')
data.head()

,Order_ID,Order_Date,Customer_ID,Customer_Name,Product_ID,Product_Name,Category,Region,Quantity,Unit_Price,Discount_%,Revenue,Payment_Mode
0,100000,2023-10-09,C0502,Customer_458,P111,Backpack,Accessories,South,2,45,0,90.0,Cash
1,100001,2024-03-08,C0066,Customer_62,P111,Backpack,Accessories,North,2,45,20,72.0,Card
2,100002,2023-07-23,C1117,Customer_860,P104,Desk,Furniture,South,1,220,20,176.0,UPI
3,100003,2023-06-13,C0866,Customer_697,P110,Mouse,Electronics,East,5,25,0,125.0,Card
4,100004,2023-04-05,C0779,Customer_199,P104,Desk,Furniture,East,6,220,0,1320.0,Online


In [5]:
data.shape

(5000, 13)

In [6]:
data['Product_Name'].nunique()

12

In [8]:
print(data['Revenue'].sum())

3753744.25


The above dataset contains 5000 sales records with 13 columns across 12 products name and total revenue is $3,753,744.25.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

1. I can claim:

      - The dataset shows which products generated the highest revenue.
      - I can compare product performance within this dataset.

2. I cannot claim:

      - These products will always be the best sellers.
      - Revenue differences prove that one product is inherently better.
      - The analysis proves why customers bought a product.

3. Remember:

      - Correlation or observation is not causation.

## Self-check

Before you submit, confirm each line honestly:

- ✅ Every section above is filled — markdown thinking AND the code that backs it
- ✅ The notebook runs top to bottom with no errors (Runtime → Run all)
- ✅ No client names, URLs, or private queries anywhere
- ✅ My claims use careful words: observed, measured, directional, decision-support
- ✅ Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.